Backup: Integrate connection error React component

chrisbliss18 opened the pull request:
#### Changes proposed in this Pull Request:
Integrate the `ConnectionErrorNotice` React component into Backup. This is part of a bigger project (1202697549597858-as) to provide consistent failed connection error reporting and an interface to reconnect (1202697549597858-as-1202697549597864) to all plugins and projects in the monorepo.

The changes will affect the following:
* Backup

#### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?

#### Jetpack product discussion
* p9dueE-5w8-p2

#### Does this pull request change what data or activity we track or use?
* No

#### Testing instructions:
1. Run `jetpack build plugins/backup`.
1. Enable the "Jetpack Debug Tools" plugin.
1. Go to Jetpack Debug > Jetpack Debug.
1. Enable "Broken token Utilities" and click "Save".
1. Go to Jetpack > Backup.
1. Confirm that no errors display below the "Jetpack" heading.
1. Go to Jetpack Debug > Broken Token.
1. Click the "Set invalid blog token" button and the "Set invalid user tokens" button.
1. Run the following query on the database to clear connection error-related transients (if running the docker Jetpack environment, run `jetpack docker db`):
```delete from wp_options where option_name like '%jetpack_connection_error_reporting_gate_%';```
1. Go to Jetpack > My Jetpack.
1. Go to Jetpack > Backup.
1. You should see an error message as shown in the below screenshot:
![jetpack-connection-error-example-generic](https://user-images.githubusercontent.com/56307/193151665-e2cee674-2084-4d5c-9054-6d1313bb447a.png)
1. Click on "Restore Connection".
1. The "Reconnecting Jetpack" message appears with the animating wait image.
1. The "Reconnecting Jetpack" connection disappears.
1. You are taken to the Jetpack connection page.

chrisbliss18 committed:
Add: Connection error to Backup

Integrate the new connection error message React component into the Backup
plugin.

chrisbliss18 committed:
changelog

chrisbliss18 committed:
Merge branch 'trunk' into add/backup-plugin-connection-errors

chrisbliss18 committed:
Merge branch 'trunk' into add/backup-plugin-connection-errors

sergeymitr commented:
Hi @chrisbliss18.
It looks like the PR needs a trunk merge and possibly a version bump.

Also, the layout might not work out:
<img width="1133" alt="Screen Shot 2022-10-04 at 13 34 44" src="https://user-images.githubusercontent.com/1341249/193887823-7e009ee3-9db4-4c40-8ce2-01f269cf3019.png">
May be we should display the notice a big higher, above the "grey background" area.
Please reach out to the plugin or design team for their input.

fgiannar committed:
Merge branch 'trunk' into add/backup-plugin-connection-errors

sergeymitr commented on projects/packages/backup/src/js/components/Admin.js:
```diff
@@ -46,6 +48,11 @@ const Admin = () => {
 					<AdminSectionHero>
 						<Container horizontalSpacing={ 0 }>
+							{ hasConnectionError && (
+								<Col>
```
Hi @chrisbliss18.
I had a similar issue on another PR, so I added the margin via styles: https://github.com/Automattic/jetpack/pull/26733/files#diff-5ab291fbb54c538c26bd2043a4ecf1eaba9e469be850f8795aa40360874458b1R217

chrisbliss18 replied:
Thanks @sergeymitr! This is exactly what I needed. I just pushed up a fix using a similar technique.

chrisbliss18 committed:
[not verified] Fix: Added missing vertical margins on error msg

The new connection error message component needed top and bottom margins
added to separate it properly from its enclosing container.

chrisbliss18 committed:
Merge branch 'trunk' into add/backup-plugin-connection-errors

chrisbliss18 committed:
fixed lock file issue

chrisbliss18 committed:
Merge branch 'trunk' into add/backup-plugin-connection-errors

sergeymitr reviewed (approved):
👍
